static/app/components/hovercard.tsx

```diff
@@ -47,11 +52,33 @@ type UseOverOverlayState = ReturnType<typeof useHoverOverlay>;
 interface HovercardContentProps
   extends Pick<
     HovercardProps,
-    'bodyClassName' | 'className' | 'header' | 'body' | 'tipColor' | 'tipBorderColor'
+    | 'animated'
+    | 'bodyClassName'
+    | 'className'
+    | 'header'
+    | 'body'
+    | 'tipColor'
+    | 'tipBorderColor'
   > {
   hoverOverlayState: Omit<UseOverOverlayState, 'isOpen' | 'wrapTrigger'>;
 }
 
+interface HovercardProviderValue {
+  isOpen: boolean;
+  reset: () => void;
+  update: (() => Promise<Partial<State>>) | null;
+}
+
+const HovercardContext = createContext<HovercardProviderValue>({
```

Comment:
Added a context here so that children can call `reset` to close the overlay when needed. This allows me to dismiss the overlay when a link is clicked.